### ISSUE TITLE
Migrating a varchar to a bigint causes invalid charset instructions to appear

### ISSUE THREAD

**Issue by JamesLF** [laravel/framework#30539]
- Laravel Version: 6.5
- PHP Version: 7.3.11
- Database Driver & Version: MySQL - 5.7.27-0ubuntu0.18.04.1

### Description:
While writing a migration to fix a bad database that had stored a foreign key as a varchar, I wrote the following migration:
```php
Schema::table('jobseeker_qualifications', function (Blueprint $table) {
        $table->bigIncrements('id')->change();
        $table->unsignedBigInteger('user_id')->change();
        $table->unsignedBigInteger('jobseekers_education_id')->change();
        $table->foreign('user_id')->references('id')->on('users');
        $table->foreign('jobseekers_education_id')->references('id')->on('jobseeker_educations');
});
```
However, this above migration generated the following SQL:
```sql
ALTER TABLE jobseeker_qualifications
CHANGE id id BIGINT UNSIGNED AUTO_INCREMENT NOT NULL,
CHANGE jobseekers_education_id jobseekers_education_id BIGINT UNSIGNED CHARACTER SET utf8 NOT NULL COLLATE `utf8_unicode_ci`,
CHANGE user_id user_id BIGINT UNSIGNED NOT NULL
```
**Work Around:** Removing the instructions related to the character set and running this query manually was successful.

### Steps To Reproduce:
1. Create a table containing a VARCHAR field that contains valid id values (only numbers).
2. Run a migration similar to the above, requesting a change the VARCHAR to unsigned BIGINT.

**Comment by rimace** [comment]
What was the error message?

**Comment by JamesLF** [comment]
> What was the error message?

```SQL Error (1064): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CHARACTER SET utf8 NOT NULL COLLATE `utf8_unicode_ci`, CHANGE user_id user_id B' at line 3```

**Comment by staudenmeir** [comment]
What version of the `doctrine/dbal` package are you using (run `composer show)`?

**Comment by JamesLF** [comment]
> What version of the doctrine/dbal package are you using (run composer show)?

Here:
`doctrine/dbal                         v2.10.0`

**Comment by staudenmeir** [comment]
Can you try downgrading to `2.9.3`?

It looks like this is the same bug: https://github.com/doctrine/dbal/issues/3714

**Comment by JamesLF** [comment]
> Can you try downgrading to 2.9.3?
> It looks like this is the same bug: doctrine/dbal#3714

It does indeed look like the same bug, I'll subscribe to that bug tracker to see when it's been fixed and I could update this ticket in turn.

I don't have the time at the moment to try the downgrade due to work pressures and the manual script work around is functional for the moment, but if I get a chance to I shall look in to it.

**Comment by driesvints** [comment]
Seems like this is related to DBAL and not Laravel so going to close this one.

**Comment by Justin991q** [comment]
DBAL just closed the issue: see https://github.com/doctrine/dbal/issues/3714#issuecomment-558573089

**Comment by JohnyProkie** [comment]
Can please folks from Laravel and folks from Doctrine make an agreement and push this issue a bit further? 
https://github.com/doctrine/dbal/issues/3714#issuecomment-559585932

Laravel says it's issue on Doctrine's side. Doctrine says it's issue on Laravel side. Meanwhile we are locked on older version of Doctrine when we want to keep our projects up-to-date.
Thank you in advance.

Please note this happened to us when changing varchar column to integer.

**Comment by lcobucci** [comment]
Hello everyone, I'm one of the core members of the Doctrine team 👋

As I mentioned on the issue in DBAL, I completely understand your frustration and am sorry for this inconvenience.

We really believe that the issue is related to how Laravel configures the DBAL objects to perform the comparisons and think that it was working previously due to a bug that got fixed. 

I don't know Laravel so much to solve this but feel free to ping me in a PR or on Doctrine's slack channel and I'll do my best to support the resolution of this issue. 

@JohnyProkie thanks for pointing us here and @driesvints and other Laravel maintainers for their hard work on building bridges between the two projects 🤟

**Comment by JohnyProkie** [comment]
@lcobucci Thank you a lot for stepping in!

**Comment by driesvints** [comment]
Let's figure this out. If anyone could help pinpoint where we can implement a fix that'd be great. 

@lcobucci thanks for your hard work on Doctrine as well :)

**Comment by lcobucci** [comment]
@driesvints I haven't debugged anything but https://github.com/laravel/framework/blob/1bbe5528568555d597582fdbec73e31f8a818dbc/src/Illuminate/Database/Schema/Grammars/ChangeColumn.php#L125 might give us a clue.

Laravel should make sure to not set the charset/collation info on the objects while performing the changes.

**Comment by driesvints** [comment]
@lcobucci it does seem that only applies to `json` and `binary` columns while the issue here is changing a column type from `varchar` to `bigint` so I'm not sure if that's the actual culprit?

https://github.com/laravel/framework/blob/1bbe5528568555d597582fdbec73e31f8a818dbc/src/Illuminate/Database/Schema/Grammars/ChangeColumn.php#L124-L128

**Comment by lcobucci** [comment]
@driesvints that's the problem. As [@AlterTable explained it](https://github.com/doctrine/dbal/issues/3714#issuecomment-549767693):

> But when it's original type was varchar/text, there is a implicit change: the charset definition should be removed. Neither Laravel nor Doctrine handled this.
> I think there are two ways to fix this, one is add extra checks in \Doctrine\DBAL\Schema\Table::changeColumn, another is modify Laravel migrate, add checks after changed attributes are set. I prefer the second one.

Laravel should define `characterset` and `collation` as `null` to the types.

Does this help?

**Comment by driesvints** [comment]
@lcobucci heya, it does. I think this is indeed the correct way to go forward. I currently don't have time to work on this so if anyone is experiencing this issue and wants to move this forward, feel free to send in a PR.

**Comment by yahya09** [comment]
Hi all, if anyone on Laravel 6.x & doctrine/dbal 2.10 experiencing this issue, the quick fix is to set charset and collation to empty string on column change statement. For example:

`$table->integer('resource_type')->default(0)->charset('')->collation('')->change();`

It does work on my case where previous column type is string (varchar).
Looks like someone has already sent PR to fix this, but I hope solution above still help before the PR merged.

**Comment by squatto** [comment]
> Hi all, if anyone on Laravel 6.x & doctrine/dbal 2.10 experiencing this issue, the quick fix is to set charset and collation to empty string on column change statement. For example:
> 
> `$table->integer('resource_type')->default(0)->charset('')->collation('')->change();`
> 
> It does work on my case where previous column type is string (varchar).
> Looks like someone has already sent PR to fix this, but I hope solution above still help before the PR merged.

This unfortunately isn't working for me for some reason. I've tried `null` and `''` for charset and collation and I still get the same error. I'm going from `string` to `bigInteger`:

```php
$table->bigInteger('number')
      ->charset('')
      ->collation('')
      ->change();
```

**Comment by taylorotwell** [comment]
Fixed by https://github.com/laravel/framework/commit/fccdf7c42d5ceb50985b3e8243d7ba650de996d6

**Comment by osbre** [comment]
@strtz try to update dependencies

**Comment by vvannak** [comment]
> Fixed by [fccdf7c](https://github.com/laravel/framework/commit/fccdf7c42d5ceb50985b3e8243d7ba650de996d6)

It worked thanks @taylorotwell
